### DISCUSSION TITLE
Add option for Router's routeChangeComplete to execute on initial load as well

### DISCUSSION THREAD

**Discussion by plug-n-play** [vercel/next.js#18092] [Ideas]
# Feature request

## Is your feature request related to a problem? Please describe.

Analytics is a pretty common use case for modern apps. A frequent implementation for Analytics is when user loads a particular screen. NextJS's Router has excellent set of lifecycle events. The most close to this use case is `routeChangeComplete` We can use this to log Analytics centrally from the application template `_app.ts` from one central location. However this doesn't works for the first page load but on all pages post the first page load.

## Describe the solution you'd like
We can have an optional flag to execute the callback of `routeChangeComplete` which would execute on first load as well.

## Describe alternatives you've considered

useEffect( () => makeAnalyticsCall(), [] ) is not clean since this needs to be applied on all pages of the app. `routeChangeComplete` is near perfect baring the first page load.

**Comment by lfades** [comment]
`routeChangeComplete` handles notifies abou route changes and that doesn't need to change or be extended.

To do something when the a page loads you should use `useEffect` in the page, or in a [custom App](https://nextjs.org/docs/advanced-features/custom-app) to do something in all pages.
